agent/php_memcached.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef PHP_MEMCACHED_HDR
+#define PHP_MEMCACHED_HDR
+
+#include "nr_datastore_instance.h"
+#include "php_includes.h"
+
+/*
+ * Purpose : Create a datastore instance metadata for a Memcached server.
+ *
+ * Params  : 1. The memcached host or socket name as given to Memcached::addServer().
+ *           2. The memcached port as given as given to Memcached::addServer().
+ *
+ * Returns: nr_datastore_instance_t* that the caller is responsible for freeing
+ */
+nr_datastore_instance_t* nr_php_memcached_create_datastore_instance(
+    const char* host_or_socket,
+    zend_long port);
+
+/*
+ * Purpose : Create a memcached instance metric
+ *
+ * Params  : 1. The memcached host or socket name as given to Memcached::addServer().
+ *              Must be non-null.
```

Comment:
Since this function is a public API, this cannot be guaranteed and therefore this function needs to account for invalid input.

Reply:
removed this comment in [c8bdfb7](https://github.com/newrelic/newrelic-php-agent/pull/958/commits/c8bdfb78b275c8e6ff9faff01a709d1c5853bc00). The function can handle NULL